### PR TITLE
dbeaver/pro#2112 show only models with priority

### PR DESCRIPTION
**Pull request by LonwoLonwo**
aux PR

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java [hunk 1]
```diff
@@ -42,6 +42,27 @@ public class DataSourceAuthModelDescriptor extends DataSourceBindingDescriptor i
 
     public static final String EXTENSION_ID = "org.jkiss.dbeaver.dataSourceAuth"; //$NON-NLS-1$
 
+    public enum DSPriority {
+        EXTRA_HIGH(1), // For necessary models
+        HIGH(2); // For priority models
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: It looks strange. If high priority exists, there should also be normal priority somewhere

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: And it is for the future. We can mark all our drivers with priority marks, but why do we need to do it without a special request?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: I just don't like the naming, that there are only high and extra high priorities in the enum

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: Ok, what do you suggest?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: I would do it this way:
Low = 0, Normal = 1, Medium = 2, High = 3

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceAuthModelDescriptor.java]: Or
Normal = 1, Medium = 2,  High = 3, Urgent = 4

**Review by ShadelessFox — COMMENTED** [review]
This approach looks a bit weird. I think there should be other ways.

**Comment by LonwoLonwo** [comment]
Approach changed.

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java [hunk 2]
```diff
@@ -109,6 +110,7 @@ public DataSourceProviderDescriptor(DataSourceProviderRegistry registry, IConfig
             this.treeDescriptor = this.loadTreeInfo(trees[0]);
         }
         this.supportsDriverMigration = CommonUtils.toBoolean(config.getAttribute("supports-migration"));
+        this.skipParentsAuthModels = CommonUtils.toBoolean(config.getAttribute("skipParentsAuthModels"));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: Please rename to `skipParentAuthModels`

Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: I'd prefer `inheritAuthModels` 😎

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ public DataSourceProviderDescriptor(DataSourceProviderRegistry registry, IConfig
             this.treeDescriptor = this.loadTreeInfo(trees[0]);
         }
         this.supportsDriverMigration = CommonUtils.toBoolean(config.getAttribute("supports-migration"));
-        this.skipParentsAuthModels = CommonUtils.toBoolean(config.getAttribute("skipParentsAuthModels"));
+        this.inheritAuthModels = CommonUtils.toBoolean(config.getAttribute("inheritAuthModels"));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: I believe the new attribute name means the opposite thing - so it should be `true` by default.

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java [hunk 4]
```diff
@@ -241,7 +241,7 @@ public DataSourceProviderDescriptor getParentProvider() {
     @Override
     public boolean matchesId(String id) {
         if (id.equals(this.id)) return true;
-        if (skipParentsAuthModels) {
+        if (inheritAuthModels) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: And here, it should be inverted

**Review by ShadelessFox — APPROVED** [review]

**Pull request merged**